### PR TITLE
Gitlab ci job

### PR DESCRIPTION
**Pull request by vlussenburg**
Screenshot 2023-01-31 at 9 11 24 AM

### CODE REVIEW

File: .gitlab-ci.yml [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+    - docker push ${REGISTRY}/lighttpd:${CI_COMMIT_REF_NAME}
+    - echo "Compile complete."
+
+mcode-test-job:   # This job runs in the test stage.
+  stage: test    # It only starts when the job in the build stage completes successfully.
+  script: 
+    - export PATH=${PATH}:~/bin   
+    - REGISTRY=$(mayhem-$(arch) docker-registry)
+    # Run Mayhem
+    # removed --merge-base-branch-name ${CI_MERGE_REQUEST_TARGET_BRANCH_NAME}
```
Comment (ethan42) [inline .gitlab-ci.yml:34]: comment cleanup?

File: .gitlab-ci.yml [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+  script:
+    # Setup aarch64 (preinstalled) and x86_64 (download to install)
+    - mkdir -p ~/bin
+    - export PATH=${PATH}:~/bin
+    - curl --no-progress-meter -Lo ~/bin/mayhem-x86_64 ${MAYHEM_URL}/cli/Linux/mayhem  && chmod +x ~/bin/mayhem-x86_64
+
+    # Login to mayhem and docker
+    - mayhem-$(arch) login --url ${MAYHEM_URL} --token ${MAYHEM_TOKEN}
+    - REGISTRY=$(mayhem-$(arch) docker-registry)
+    - echo "${MAYHEM_TOKEN}" | docker login -u ${MAYHEM_USERNAME} --password-stdin ${REGISTRY}
```
Comment (ethan42) [inline .gitlab-ci.yml:17]: possible you can do it with --prompt / MAYHEM_PROMPT?

```
  --prompt              Force prompt text to display. Useful when no 
                        TTY/interactive shell is connected, such as in a CI/CD pipeline,
                        and a user wants to control prompt input. Equivalent 
                        to setting the environment variable `MAYHEM_PROMPT=1`.
```

File: .gitlab-ci.yml [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+
+    # Determine run name
+    - runName=$(echo ${run} | awk -F / '{ print $(NF-1) }');
+
+    # Wait for job to complete and artifacts to be ready
+    - mayhem-$(arch) --verbosity info wait $run --owner forallsecure --sarif sarif-${runName}.sarif --junit junit-${runName}.xml;
+    - status=$(mayhem-$(arch) --verbosity info show --owner forallsecure --format json $run | jq '.[0].status')
+    - if [[ ${status} == *"stopped"* || ${status} == *"failed"* ]]; then exit 2; fi
+    - defects=$(mayhem --verbosity info show --owner forallsecure --format json ${run} | jq '.[0].defects|tonumber')
+    - if [[ ${defects} -gt 0 ]]; then echo '${defects} defects found!'; exit 3; fi
```
Comment (ethan42) [inline .gitlab-ci.yml:50]: I'd consider using the security dashboard or the junit reporting for blocking here, but probably a matter of preference.

File: .gitlab-ci.yml (hunk 1 above)
Comment (vlussenburg) [reply .gitlab-ci.yml:34]: Ideally I would want to re-add these removed flags before merging the PR :)

File: .gitlab-ci.yml (hunk 3 above)
Comment (vlussenburg) [reply .gitlab-ci.yml:50]: What do you mean by security dashboard here? Can I use that non-interactively?

jUnit reporting is another good option preventing a network request. I didn't find good options how however:
https://forum.gitlab.com/t/how-to-fail-job-and-pipeline-if-there-are-failed-tests-in-junit-report/37039

Do you have any bash one liners I could use here?

File: .gitlab-ci.yml (hunk 2 above)
Comment (vlussenburg) [reply .gitlab-ci.yml:17]: Will try. Thanks.

Comment (vlussenburg) [reply .gitlab-ci.yml:17]: Works!


```
$ MAYHEM_PROMPT=1 mayhem-$(arch) login --url ${MAYHEM_URL} --token ${MAYHEM_TOKEN}
[21](http://localhost:8080/root/mcode-action-examples/-/jobs/34#L21)WARNING: The API and Client versions mismatch, consider upgrading (client: 2.3.0, API: 2.2.0). Continue at your own risk.
[22](http://localhost:8080/root/mcode-action-examples/-/jobs/34#L22)Logged in successfully at '[MASKED]:443' as 'vlussenburg'.
[23](http://localhost:8080/root/mcode-action-examples/-/jobs/34#L23)Syncing default settings: /home/gitlab-runner/.config/mayhem/mayhem.
[24](http://localhost:8080/root/mcode-action-examples/-/jobs/34#L24)Logging into the Docker registry at demo.forallsecure.com:5000
[25](http://localhost:8080/root/mcode-action-examples/-/jobs/34#L25)Already logged in with the same token
```

**Review by unionfindbee — APPROVED** [review]
Looks good!

**Pull request merged**